### PR TITLE
Conversion of user ID from an integer to a string, with it's own doma…

### PR DESCRIPTION
**Pull request by williamthurley**
…in class.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -3,4 +3,18 @@ version := "0.1"
 scalaVersion := "2.12.4"
 
 // https://mvnrepository.com/artifact/uk.co.jhc/sqlest
-libraryDependencies += "uk.co.jhc" %% "sqlest" % "0.8.10"
+libraryDependencies ++= Seq(
+  "com.typesafe" % "config" % "1.3.0",
+  "uk.co.jhc" %% "sqlest" % "0.8.8",
```
Comment (ben-mulhern) [inline build.sbt]: Has reverted the version slightly, I'd find the latest version from the Github page,

File: build.sbt [hunk 2]
```diff
@@ -3,4 +3,18 @@ version := "0.1"
...
+  "org.json4s" %% "json4s-native" % "3.5.1",
+  "org.json4s" %% "json4s-ext" % "3.5.1",  
+  "com.typesafe.scala-logging" %% "scala-logging" % "3.5.0",
+  "org.scalatest" %% "scalatest" % "3.0.1" % "test",
+  "org.scalactic" %% "scalactic" % "3.0.1",
+  "com.h2database" % "h2" % "1.4.191",
+  "org.http4s" %% "http4s-dsl" % "0.15.8",
+  "org.http4s" %% "http4s-blaze-server" % "0.15.8"
```
Comment (ben-mulhern) [inline build.sbt]: The only ones we need right now are the com.typesafe config one, the sqlest one, and the h2database one. I'd ditch the rest.

File: src/main/scala/dal/PasswordSaltExtractor.scala [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package dal
```
Comment (ben-mulhern) [inline src/main/scala/dal/PasswordSaltExtractor.scala]: We established on Slack that the extractor wasn't required at all.

File: src/main/scala/dal/SqlestDb.scala [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package dal
+
+import sqlest._
+import com.typesafe.config._
+//import server.Config._
```
Comment (ben-mulhern) [inline src/main/scala/dal/SqlestDb.scala]: This is meant to be fetching the data out of the application.conf text file. We need this for the url, username and password of the database. I will provide some extra info about this to Sean,

File: src/main/scala/dal/SqlestDb.scala [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+trait SqlestDb {
+
+  val dataSource = {
+    val dataSource = new org.h2.jdbcx.JdbcDataSource
+//    dataSource.setURL(url)
+//    dataSource.setUser(user)
+//    dataSource.setPassword(password)
```
Comment (ben-mulhern) [inline src/main/scala/dal/SqlestDb.scala]: Reinstate these three once we've sorted the config retrieval.

File: src/main/scala/dal/UserDal.scala [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+
+  def getPasswordSalt(id: UserID): Option[(String,String)]  = {
+
+  	val query =
+      select
+        .from(UserDetailTable)
+        .where(UserDetailTable.userID == id)
+
+    query.extractHeadOption(PasswordSaltExtractor)
```
Comment (ben-mulhern) [inline src/main/scala/dal/UserDal.scala]: See Slack conversation, we suggested something like this:

      val query =
      select(UserDetailTable.password, UserDetailTable.salt)
        .from(UserDetailTable)
        .where(UserDetailTable.userID == id)

    query.fetchHeadOption

File: src/main/sql/user_detail.sql [hunk 7]
```diff
@@ -4,9 +4,8 @@ CREATE TABLE user_detail
   PRIMARY KEY (user_id),                                             
                                                                          
 --Columns                                                              
-  user_id INT NOT NULL GENERATED ALWAYS AS IDENTITY,
+  user_id VARCHAR(50) NOT NULL,
```
Comment (ben-mulhern) [inline src/main/sql/user_detail.sql:7]: This is fine but we'll need to update the actual database at some point.

File: build.sbt (hunk 2 above)
Comment (williamthurley) [reply build.sbt]: Done

File: src/main/scala/dal/PasswordSaltExtractor.scala (hunk 3 above)
Comment (williamthurley) [reply src/main/scala/dal/PasswordSaltExtractor.scala]: Done

File: src/main/scala/dal/UserDal.scala (hunk 6 above)
Comment (williamthurley) [reply src/main/scala/dal/UserDal.scala]: Done

**Pull request merged**